client/src/app/site/assignments/components/assignment-poll/assignment-poll-dialog.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA, MatSnackBar } from '@angular/material';
+import { TranslateService } from '@ngx-translate/core';
+
+import { AssignmentPollService } from '../../services/assignment-poll.service';
+import { CalculablePollKey } from 'app/core/ui-services/poll.service';
+import { Poll } from 'app/shared/models/assignments/poll';
+import { PollOption, PollVoteValue } from 'app/shared/models/assignments/poll-option';
+import { ViewUser } from 'app/site/users/models/view-user';
+
+/**
+ * A dialog for updating the values of an assignment-related poll.
+ */
+@Component({
+    selector: 'os-assignment-poll-dialog',
+    templateUrl: './assignment-poll-dialog.component.html',
+    styleUrls: ['./assignment-poll-dialog.component.scss']
+})
+export class AssignmentPollDialogComponent {
+    /**
+     * List of accepted special non-numerical values from {@link PollService}
+     */
+    public specialValues: [number, string][];
+
+    /**
+     * vote entries for each option in this component. Is empty if method
+     * requires one vote per candidate
+     */
+    public optionPollKeys: PollVoteValue[];
+
+    /**
+     * Vote entries included once for summary (e.g. total votes cast)
+     */
+    public summaryPollKeys: CalculablePollKey[];
+
+    /**
+     * Constructor. Retrieves necessary metadata from the pollService,
+     * injects the poll itself
+     */
+    public constructor(
+        public dialogRef: MatDialogRef<AssignmentPollDialogComponent>,
+        @Inject(MAT_DIALOG_DATA) public data: { poll: Poll; users: ViewUser[] },
+        private matSnackBar: MatSnackBar,
+        private translate: TranslateService,
+        private pollService: AssignmentPollService
+    ) {
+        this.specialValues = this.pollService.specialPollVotes;
+        switch (this.data.poll.pollmethod) {
+            case 'votes':
+                this.optionPollKeys = ['Yes'];
+                break;
+            case 'yn':
+                this.optionPollKeys = ['Yes', 'No'];
+                break;
+            case 'yna':
+                this.optionPollKeys = ['Yes', 'No', 'Abstain'];
+                break;
+        }
+    }
+
+    /**
+     * Close the dialog, submitting nothing. Triggered by the cancel button and
+     * default angular cancelling behavior
+     */
+    public cancel(): void {
+        this.dialogRef.close();
+    }
+
+    /**
+     * Validates candidates input (every candidate has their options filled in),
+     * submits and closes the dialog if successfull, else displays an error popup.
+     * TODO better validation
+     */
+    public submit(): void {
+        const error = this.data.poll.options.find(dataoption => {
+            for (const key of this.optionPollKeys) {
+                const keyValue = dataoption.votes.find(o => o.value === key);
+                if (!keyValue || keyValue.weight === undefined) {
```

Comment:
`return !keyValue`. This is also true for `undefined`.

Reply:
This is another cultural thingy where I might be using outdated code. The intention is: 'check all data and abort (return true) as soon as there is anything fishy'. So 'false' should only return if none of the checks is true.